### PR TITLE
implement api-gateway module

### LINKED ISSUE #15: implement @aloxide/api-gateway module
ref #6 

One of the factors to affect dApp's performance and user experience is how the client-side application fetches data stored in the smart contract. Smart contracts don't support sophisticated queries unlike typical data stores like cache, NoSQL, and RDBMS, not to mention performance.

> Aloxide API gateway is a standalone application server that provides generic-purpose CRUD operations in GraphQL as well as mirror tables on the centralized database for Aloxide models mapped to the smart contract table.

Let's build a dApp without constraints of the smart contract table using the API gateway just like we develop typical server-client applications.

### PR DESCRIPTION
**Pull request by manh-vv**
close #15 

implemented module:

* **@aloxide/model**: this model is for generating sequelize models which are then used in demux and api-gateway
* **@aloxide/demux**: follow demux pattern. Note that in this MR only EOS was implemented.
* **@aloxide/api-gateway**: a bootstrapping for setting up node server
* add example of using api-gateway and demux: `example-api-gateway`.

For demo: https://github.com/lecle/aloxide/tree/feat/15-implement-api-gateway-module/packages/example-api-gateway#start-demo

### CODE REVIEW

File: packages/example-api-gateway/src/loadEnv.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  if (process.env.dote_loaded) {
+    return;
+  }
+
+  dote.load({
+    encoding: 'utf8',
+    silent: true,
+    path: '.env',
+    defaults: '.env.defaults',
+    schema: '.env.defaults',
```
Comment (quocle108) [inline packages/example-api-gateway/src/loadEnv.ts:18]: it should be .env, right?

File: packages/example-api-gateway/src/readBlockchain.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  accountName: 'helloworld12',
+  modelBuilderConfig: {
+    aloxideConfigPath: config.aloxideConfigPath,
+    logger: config.logger,
+  },
+  sequelize: config.sequelize,
+  nodeActionReaderOptions: {
+    nodeosEndpoint: process.env.app_nodeosEndpoint,
+    onlyIrreversible: false,
+    startAtBlock: 9130000,
```
Comment (quocle108) [inline packages/example-api-gateway/src/readBlockchain.ts:15]: the start block should be loaded from constants file or env file?

File: packages/example-api-gateway/src/readBlockchain.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { createWatcher } from '@aloxide/demux';
+
+import config from './config';
+
+createWatcher({
+  accountName: 'helloworld12',
```
Comment (quocle108) [inline packages/example-api-gateway/src/readBlockchain.ts:6]: load from env

File: packages/example-api-gateway/src/loadEnv.ts (hunk 1 above)
Comment (manh-vv) [reply packages/example-api-gateway/src/loadEnv.ts:18]: No, it is correct.

File: packages/example-api-gateway/src/dummy.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+const api = new Api({
+  rpc,
+  signatureProvider,
+  textDecoder: new TextDecoder(),
+  textEncoder: new TextEncoder(),
+});
+
+const eosAccountName = process.env.app_d_eos_account_name || 'helloworld12';
+
+async function cleanUpEos() {
```
[the remark this answers is not in the document]
Comment (manh-vv) [reply packages/example-api-gateway/src/dummy.ts:20]: Right, it is an example. I will remove it.

File: packages/api-gateway/src/express-router.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+      const model = sequelize.models[name];
+      model.findByPk(req.params.id).then(item => {
+        res.send({
+          item,
+        });
+      });
+    });
+
+    expressRouter.get('/', (req, res) => {
+      const reDocHtml = `<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (manh-vv) [reply packages/api-gateway/src/express-router.ts:68]: There is no need for now.

File: packages/demux/src/index.ts [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  modelBuilder.build(sequelize);
+
+  const aloxideConfig = modelBuilder.aloxideConfig;
+
+  if (!actionHandler) {
+    if (!handlerVersion) {
+      handlerVersion = new BaseHandlerVersion(
+        versionName,
+        aloxideConfig.entities
+          .map<AbsDbUpdater[]>(({ name }) => {
```
[the remark this answers is not in the document]
Comment (manh-vv) [reply packages/demux/src/index.ts:58]: I think you meant support model versioning. If so it is a good idea. I will create a new issue and mark it as a feature request.

File: packages/example-api-gateway/src/readBlockchain.ts (hunk 2 above)
Comment (manh-vv) [reply packages/example-api-gateway/src/readBlockchain.ts]: Fixed!

File: packages/example-api-gateway/src/readBlockchain.ts (hunk 3 above)
Comment (manh-vv) [reply packages/example-api-gateway/src/readBlockchain.ts]: Fixed!

File: packages/demux/src/index.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (quocle108) [reply packages/demux/src/index.ts:58]: Yes, right

**Pull request merged**